src/Workspaces/Core/Portable/Options/GlobalOptionService.cs

```diff
@@ -153,22 +125,11 @@ public void SetOptions(OptionSet optionSet)
 
                     _currentValues = _currentValues.SetItem(optionKey, setValue);
 
-                    ImmutableArray<Lazy<IOptionSerializer, OptionSerializerMetadata>> optionSerializers;
-                    if (_featureNameToOptionSerializers.TryGetValue(optionKey.Option.Feature, out optionSerializers))
+                    foreach (var serializer in _optionSerializers)
                     {
-                        foreach (var serializer in optionSerializers)
+                        if (serializer.Value.TryPersist(optionKey, setValue))
```

Comment:
The assumption is only one serializer (per scope, which is work still coming) will respond to the metadata on the option type. If you created an option with two then yes there's a problem. Since that part of the API is internal, the response is "don't do that.".